lib/onebox/helpers.rb

```diff
@@ -63,8 +63,7 @@ def self.fetch_response(location, redirect_limit: 5, domain: nil, headers: nil,
       end
 
       result = StringIO.new
-      Net::HTTP.start(uri.host, uri.port, use_ssl: uri.normalized_scheme == 'https') do |http|
-        http.open_timeout = Onebox.options.connect_timeout
+      Net::HTTP.start(uri.host, uri.port, open_timeout: Onebox.options.connect_timeout, use_ssl: uri.normalized_scheme == 'https') do |http|
```

Comment:
start will attempt to connect right away, setting the open_timeout in the block is too late